drivers/cc1200/cc1200.c

```diff
@@ -0,0 +1,361 @@
+    cc1200_write_reg(dev, CC1200_FS_VCO0, 0xB5);
+    cc1200_write_reg(dev, CC1200_IFAMP, 0x05);
+    cc1200_write_reg(dev, CC1200_XOSC5, 0x0E);
+    cc1200_write_reg(dev, CC1200_XOSC1, 0x03);
+    cc1200_write_reg(dev, CC1200_AGC_CS_THR, (uint8_t) -91);
+    cc1200_write_reg(dev, CC1200_AGC_GAIN_ADJUST, (int8_t)CC1200_RF_CFG_RSSI_OFFSET);
+    cc1200_set_channel(dev, CC1200_DEFAULT_CHANNEL);
+
+    /* set default node id */
+    uint64_t addr;
+    luid_get(&addr, 8);
+    cc1200_set_address_long(dev, addr);
+    cc1200_set_address_short(dev, (uint16_t)addr & 0xFFFF);
+    cc1200_set_address(dev, (uint8_t) addr & 0xFF);
+
+
+
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+    LOG_INFO("cc1200: initialized with address=%u and channel=%i\n",
+            (unsigned)dev->radio_address,
+            dev->radio_channel);
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+    return 0;
+}
+
+uint8_t cc1200_set_address(cc1200_t *dev, uint8_t address)
+{
+    DEBUG("%s:%s:%u setting address %u\n", RIOT_FILE_RELATIVE, __func__,
+            __LINE__, (unsigned)address);
+    if (!(address < MIN_UID) || (address > MAX_UID)) {
+        if (dev->radio_state != RADIO_UNKNOWN) {
+
+            cc1200_write_register(dev, CC1200_DEV_ADDR, address);
+            dev->radio_address = address;
+            return address;
+        }
+    }
+
+    return 0;
+}
+
+uint64_t cc1200_set_address_long(cc1200_t *dev, uint64_t address)
+{
+    DEBUG("%s:%s:%u setting address long 0x%x\n", RIOT_FILE_RELATIVE, __func__,
+            __LINE__, (unsigned)address);
+    if (!(address < MIN_UID) || (address > 0xFFFFFFFFFFFFFFFF)) {
+        if (dev->radio_state != RADIO_UNKNOWN) {
+      
+            dev->radio_address_long = address;
+            return address;
+        }
+    }
+
+    return 0;
+}
+
+uint16_t cc1200_set_address_short(cc1200_t *dev, uint16_t address)
+{
+    DEBUG("%s:%s:%u setting address short 0x%x\n", RIOT_FILE_RELATIVE, __func__,
+            __LINE__, (unsigned)address);
+    if (!(address < MIN_UID) || (address > 0xFFFF)) {
+        if (dev->radio_state != RADIO_UNKNOWN) {
+      
+            dev->radio_address_short = address;
+            return address;
+        }
+    }
+
+    return 0;
+}
+
+uint16_t cc1200_set_pan_id(cc1200_t *dev, uint16_t pan_id)
+{
+    DEBUG("%s:%s:%u setting pan id 0x%x\n", RIOT_FILE_RELATIVE, __func__,
+            __LINE__, (unsigned)pan_id);
+    if (!(pan_id < MIN_UID) || (pan_id > 0xFF)) {
+        if (dev->radio_state != RADIO_UNKNOWN) {
+      
+            dev->pan_id = pan_id;
+            return pan_id;
+        }
+    }
+
+    return 0;
+}
+
+void cc1200_set_base_freq_raw(cc1200_t *dev, const char* freq_array)
+{
+#if ENABLE_DEBUG == 1
+    uint8_t _tmp[] = { freq_array[2], freq_array[1], freq_array[0], 0x00};
+    uint32_t *FREQ = (uint32_t*) _tmp;
+
+    DEBUG("cc1200_set_base_freq_raw(): setting base frequency to %uHz\n",
+            (26000000>>16) * (unsigned)(*FREQ));
```

Comment:
I know this is from the cc110x driver but anyway,` (26000000>>16)` is not really speaking. Can it be simplified?